### ISSUE TITLE
Repeated content type headers in WS on POST

### ISSUE THREAD

**Issue by wsargent** [playframework/playframework#4681]
From https://groups.google.com/d/topic/play-framework/mF-Rqh_ljEA/discussion

wenn I do that:

```scala
WS.url(url)
     .withHeaders("x-api-version" -> "1.0")
     .withRequestTimeout(2000)
     .post(authData)
     .map { response =>
     response.status match {
       case s if s < 300 =>
         Ok(response.json)
       case _ =>
         Ok("Status: " + response.status + "/ Error: " + response.body)
     }
   }
```

I see this request:

```
Request DefaultHttpRequest(chunked: false)
POST /api/login/ HTTP/1.1
Host: 62.67.13.54
Content-Type: application/json; charset=utf-8
Connection: keep-alive
Accept: */*
User-Agent: HbbTVPlugin
Content-Length: 37
```

The the target system has a problem processing `"charset=utf-8"`, so I tried his:

```scala
WS.url(url)
     .withHeaders("x-api-version" -> "1.0")
     .withHeaders(CONTENT_TYPE -> "application/json")
     .withRequestTimeout(2000)
     .post(authData)
     .map { response =>
     response.status match {
       case s if s < 300 =>
         Ok(response.json)
       case _ =>
         Ok("Status: " + response.status + "/ Error: " + response.body)
     }
   }
```

and I get this request:

```
Request DefaultHttpRequest(chunked: false)
POST /api/login/ HTTP/1.1
Host: 62.67.13.54
Content-Type: application/json
Content-Type: application/json; charset=utf-8
Connection: keep-alive
Accept: */*
User-Agent: HbbTVPlugin
Content-Length: 37
```

I did not expect to get the same header key (`Content-Type`) twice?

What can I do to change the content type, without getting two content-type header?

**Comment by marcospereira** [comment]
This also happens when using async-http-client directly. How to reproduce:

1. Clone playframework code
2. `cd framework`
3. `sbt Play-WS/console`
4. Execute the following code inside the console

```scala
import com.ning.http.client.AsyncHttpClient

val client = new AsyncHttpClient()
client.prepareGet("http://google.com").addHeader("Content-Type", "utf-8").addHeader("Content-Type", "application/json").execute()
```

The logs will show the duplicated header:

```
scala> 01:55:05.982 [New I/O worker #1] DEBUG c.n.h.c.p.n.r.NettyConnectListener - Request using non cached Channel '[id: 0x94aabe3f, /192.168.1.89:50367 => google.com/186.192.90.5:80]':
DefaultHttpRequest(chunked: false)
GET / HTTP/1.1
Content-Type: utf-8
Content-Type: application/json
Connection: keep-alive
Host: globo.com
Accept: */*
User-Agent: AHC/1.0
```
Maybe we should raise a bug against async-http-client itself? Or perhaps `WSRequest` trait should expose a `setHeader` method which replaces the header value instead of add a new one.

@wsargent and @jroper what do you think?

**Comment by mkurz** [comment]
Update for Play 2.4.6:

Using **Java**:
```java
	WS.url(url)
		.setHeader("x-api-version", "1.0")
		.setHeader(CONTENT_TYPE, "application/json")
		.setHeader(CONTENT_TYPE, "application/xml")
		...
```
produces
```
POST / HTTP/1.1
x-api-version: 1.0
Content-Type: application/json; charset=utf-8
Content-Length: 4
Connection: keep-alive
Host: google.com
Accept: */*
User-Agent: AHC/1.0
```

Using **Scala**:
```scala
	WS.url("http://google.com")
		.withHeaders("x-api-version" -> "1.0")
		.withHeaders(CONTENT_TYPE -> "application/json")
		.withHeaders(CONTENT_TYPE -> "application/xml")
		...
```
produces
```
POST / HTTP/1.1
x-api-version: 1.0
Content-Type: application/json
Content-Type: application/xml
Content-Length: 4
Connection: keep-alive
Host: google.com
Accept: */*
User-Agent: AHC/1.0
```


So the Java and Scala versions differ. In Java the first header is used where probably the second one should be used instead (because it should override the first one). The Scala version still produces duplicate headers.

@wsargent @marcospereira Is the Java WS not using the Scala WS code within Play?

I also tried async-http-client 2.0.0-RC7 and it still fails here as well:
```java
final AsyncHttpClient client = new DefaultAsyncHttpClient();
client.prepareGet("http://google.com").addHeader("Content-Type", "utf-8").addHeader("Content-Type", "application/json").execute();
```
produces
```
GET / HTTP/1.1
Content-Type: utf-8
Content-Type: application/json
Host: google.com
Accept: */*
User-Agent: AHC/2.0
```

**Comment by mkurz** [comment]
Fix: #5477
